crates/freeze/src/datasets/transactions.rs

```diff
@@ -89,16 +103,37 @@ impl CollectByBlock for Transactions {
             } else {
                 Box::new(|_| true)
             };
+
+        // filter by address
+        let addr_filter: Box<dyn Fn(&Transaction) -> bool + Send> =
+        if let Some(address) = &request.address {
+            Box::new(move |tx| {
+                let mut transactions_set = TRANSACTIONS_SET.lock().unwrap();
```

Comment:
what is the reason for using `TRANSACTIONS_SET` here?

Reply:
In cases where you prompt 2 or more addresses, anytime both the sender and receiver of the transaction are in --addresses, you will get duplicated transactions in the df. Originally, I used TRANSACTIONS_SET to prevent duplicates.

I've since made things more efficient and avoided global variables by making the list of addresses accessible.
This is how it works now:
```
from_bool = addresses.contains(from_address)
to_bool = addresses.contains(to_address)
if from_bool and to_bool:
    return from_address==current_address
else:
    return from_address==current_address or to_address==current_address
```
I'm unsure if the way I'm getting the list of addresses is the best way, but I couldn't figure out how else to do it. Currently, it gets the addresses by parsing 'cli_command' from ExecutionEnv.

Reply:
Thanks! I can't seem to find `.addresses()`, in request. There is  `.address` though, which only has one address :/ 
Since cryo processes filters transactions by each address separately, I'd need the complete list of addresses in addition to the address for the current chunk to prevent duplicates.